kore/src/Kore/Step/Rule.hs

```diff
@@ -568,15 +568,17 @@ refreshRulePattern (FreeVariables.getFreeVariables -> avoid) rule1 =
         left' = TermLike.substitute subst left
         right' = TermLike.substitute subst right
         requires' = Predicate.substitute subst requires
+        ensures' = Predicate.substitute subst ensures
         rule2 =
             rule1
                 { left = left'
                 , right = right'
                 , requires = requires'
+                , ensures = ensures'
                 }
     in (rename, rule2)
   where
-    RulePattern { left, right, requires } = rule1
+    RulePattern { left, right, requires, ensures } = rule1
```

Comment:
Could you please split the bugfix into its own pull request and add a test?

Reply:
https://github.com/kframework/kore/pull/967